Automattic/wp-calypso
# Add a handler to API error when submitting the ticket on the Migrate Message step

Related to #

## Proposed Changes

* We noticed that we were not handling the errors when trying to submit the ticket via the Migrate Message step.
* I added a simple message when the request fails.
* We also didn't have track events for success or failure, I added the following events:
  *  calypso_importer_migration_ticket_submit_success
  * calypso_importer_migration_ticket_submit_error
* I also had to update the revamp flow to reflect the change we made.

![Captura de Tela 2024-09-09 às 23 05 53](https://github.com/user-attachments/assets/d6a7cb24-b3f8-4558-9005-b7c1ebc1c5c1)

In a follow-up PR, I will also update the flow for the import-hosted-site flow.


## Testing Instructions



First we need to make the ticket creation endpoint return an error, so we can see the error message.
* Make sure you are sandboxed
* Edit the method below to always return an error:
fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Serfg%2Qncv%2Qcyhtvaf%2Sraqcbvagf%2Suryc%2Qzvtengvba%2Qgvpxrg.cuc%3Se%3Q487565o5%2361-og
```
return new WP_Error('test','message');
```

* Now go through the migration flow by navigating to `/start`:
  * Go through the domain selection step
  * Select the free plan on the `Plans` page
  * On the `Goals` step, select the "Import existing content or website" option and click continue
  * Input the source site URL in the Identify step
  * Next, select the "Migrate Site" option on the `Import or Migrate` step
  * Select "Do it for me" on the `How to migrate` step
  * Go through the checkout
  * Once you complete the checkout, you should land on the new `Credentials` step
  * Now skip the step, and you should be redirected to the Migrate Message step and see the error message.
  * You can also check the track events are being fire correctly
  * You can also check the successful state, by removing the error you introduced.

### Testing the revamp flow
* Navigate to `/setup/migration` and go through the assisted flow
* Don't forget to make the endpoint linked above return an error.
* Once you reach the credentials step, skip it
* You should be redirected back to the Credentials step with an error message.
* You can also undo the error on the endpoint to ensure the ticket is created and you don't see an error.

**Note:** Don't forget to at least leave a message in your ticket information that it was a test and asking the HEs to close it.

## Pre-merge Checklist
